src/code.sh

```
@@ -81,150 +44,326 @@ _trim_fastq_endings () {
-for i in "${!R1_test[@]}"; do
-       if [[ ! "${R2_test}" =~ "${R1_test[$i]}" ]]; then 
-              echo "Each R1 FASTQ does not appear to have a matching R2 FASTQ"
-              echo "${R2_test} ${R1_test[$i]}"
-              exit 1 
+_sense_check_fastq_arrays() {
+       : '''
+       Running tests to sense-check R1/R2 arrays and file prefixes
+       '''
+       # Check that there are the same number of R1s as R2s
+       cd /home/dnanexus/r1_fastqs
+       R1=($(ls *_R1_*))
```

Shellcheck (0.10.0)

[warning] Prefer mapfile or read -a to split command output (or quote to avoid splitting).

(SC2207)